Lib/ast.py

```diff
@@ -1001,11 +1001,13 @@ def _write_docstring(self, node):
         if node.kind == "u":
             self.write("u")
 
-        # Preserve quotes in the docstring by escaping them
-        value = node.value.replace("\\", "\\\\")
-        value = value.replace('"""', '""\"')
-        if value and value[-1] == '"':
-            value = value.replace('"', '\\"', -1)
+        value = node.value
+        if value:
+            # Preserve quotes in the docstring by escaping them
+            value = value.replace("\\", "\\\\")
+            value = value.replace('"""', '""\"')
```

Comment:
value = value.replace("\\", "\\\\").replace('"""', '""\"')

Reply:
Not super crazy about this, just a suggestion

Reply:
I find this version more explicit and step by step, but I'm OK if you think this one is more readable.